src/aggregation_merkle_sum_tree.rs

```diff
@@ -93,40 +124,37 @@ impl<const N_ASSETS: usize, const N_BYTES: usize> Tree<N_ASSETS, N_BYTES>
     }
 }
 
-impl<const N_ASSETS: usize, const N_BYTES: usize> AggregationMerkleSumTree<N_ASSETS, N_BYTES> {
-    /// Builds a AggregationMerkleSumTree from a set of mini MerkleSumTrees
-    /// The leaves of the AggregationMerkleSumTree are the roots of the mini MerkleSumTrees
+impl<const N_CURRENCIES: usize, const N_BYTES: usize>
+    AggregationMerkleSumTree<N_CURRENCIES, N_BYTES>
+{
+    /// Builds a AggregationMerkleSumTree from a set of mini MerkleSumTree
+    /// The leaves of the AggregationMerkleSumTree are the roots of the mini MerkleSumTree
     pub fn new(
-        mini_trees: Vec<MerkleSumTree<N_ASSETS, N_BYTES>>,
-    ) -> Result<Self, Box<dyn std::error::Error>>
+        mini_trees: Vec<MerkleSumTree<N_CURRENCIES, N_BYTES>>,
+        cryptocurrencies: Vec<Cryptocurrency>,
+    ) -> Result<AggregationMerkleSumTree<N_CURRENCIES, N_BYTES>, Box<dyn std::error::Error>>
     where
-        [usize; N_ASSETS + 1]: Sized,
-        [usize; 2 * (1 + N_ASSETS)]: Sized,
+        [usize; N_CURRENCIES + 1]: Sized,
+        [usize; N_CURRENCIES + 2]: Sized,
     {
+        if mini_trees.is_empty() {
+            return Err("Empty mini tree inputs".into());
+        }
+
         // assert that all mini trees have the same depth
         let depth = mini_trees[0].depth();
         assert!(mini_trees.iter().all(|x| x.depth() == depth));
 
-        Self::build_tree(mini_trees)
-    }
-
-    fn build_tree(
-        mini_trees: Vec<MerkleSumTree<N_ASSETS, N_BYTES>>,
-    ) -> Result<AggregationMerkleSumTree<N_ASSETS, N_BYTES>, Box<dyn std::error::Error>>
-    where
-        [usize; N_ASSETS + 1]: Sized,
-        [usize; 2 * (1 + N_ASSETS)]: Sized,
-    {
         // extract all the roots of the mini trees
         let roots = mini_trees
             .iter()
             .map(|x| x.root().clone())
-            .collect::<Vec<Node<N_ASSETS>>>();
+            .collect::<Vec<Node<N_CURRENCIES>>>();
 
         let depth = (roots.len() as f64).log2().ceil() as usize;
 
         // Calculate the accumulated balances for each asset
-        let mut balances_acc: Vec<Fp> = vec![Fp::from(0); N_ASSETS];
+        let mut balances_acc: Vec<Fp> = vec![Fp::from(0); N_CURRENCIES];
```

Comment:
Can you remove the range check on the balances performed on line 165 to 176. We removed it from `summa-solvency` here https://github.com/summa-dev/summa-solvency/pull/209/commits/bf7b16ae0b0167d92e9d7e5a2203462c814664bc#diff-6806060da92a5ec3994a7c2145b4a65b64c4d8e1217f719d9c0b3c526ea2ffcf

Also remove the correlated test as well.

Reply:
Removed at [c8a2d7f](https://github.com/summa-dev/summa-aggregation/pull/3/commits/c8a2d7f0ae2411b5e44a86837bc43dda5256f6be)